latex/chain.tex

```diff
@@ -0,0 +1,787 @@
+  This is an implementation decision.
+\end{todo}
+\clearpage
+
+\subsection{Block Definitions}
+\label{sec:defs-blocks}
+
+\begin{figure*}[htb]
+  \emph{Abstract types}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \var{hb} & \HashBlock & \text{hash of a block body}\\
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Block Header Body}
+  %
+  \begin{equation*}
+    \BHBody =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{prev} & \HashBlock & \text{hash of previous block body}\\
+        \var{hash} & \HashBlock & \text{hash of current block body}\\
+        \var{vk} & \VKey & \text{block issuer}\\
+        \var{slot} & \Slot & \text{block slot}\\
+        \var{seed_{n}} & \Seed & \text{nonce seed}\\
+        (n,~\var{prf_{n}}) & \unitInterval\times\Proof & \text{nonce proof}\\
+        \var{seed_{\ell}} & \Seed & \text{leader election seed}\\
+        (\ell,~\var{prf_{\ell}}) & \unitInterval\times\Proof & \text{leader election proof}\\
+      \end{array}
+    \right)
+  \end{equation*}
+  %
+  \emph{Block Types}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}l@{\qquad=\qquad}l}
+      \var{bh}
+      & \BHeader
+      & \BHBody \times \Sig
+      \\
+      \var{b}
+      & \Block
+      & \BHeader \times \seqof{\Tx}
+    \end{array}
+  \end{equation*}
+  \emph{Abstract functions}
+  %
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \hashBlock{} & \seqof{\Tx} \to \HashBlock
+                   & \text{hash of a block body} \\
+      \slotToSeed{} & \Slot \to \Seed
+                    & \text{convert a slot to a seed} \\
+    \end{array}
+  \end{equation*}
+  %
+  \emph{Accessor Functions}
+  \begin{equation*}
+    \begin{array}{r@{~\in~}lr}
+      \fun{header} & \Block \to \BHeader \\
+      \fun{hbody} & \Block \to \BHBody \\
+      \fun{hsig} & \Block \to \Sig\\
+      \fun{bbody} & \Block \to \seqof{\Tx} \\
+      \fun{bprev} & \BHBody \to \HashBlock\\
+      \fun{bhash} & \BHBody \to \HashBlock\\
+      \fun{bissuer} & \BHBody \to \VKey\\
+      \fun{bslot} & \BHBody \to \Slot\\
+      \fun{bseed}_n & \BHBody \to \Seed\\
+      \fun{bprf}_n & \BHBody \to \unitInterval\times\Proof\\
+      \fun{bseed}_\ell & \BHBody \to \Seed\\
+      \fun{bprf}_\ell & \BHBody \to \unitInterval\times\Proof\\
+    \end{array}
+  \end{equation*}
+  %
+  \caption{Block Definitions}
+  \label{fig:defs:blocks}
+\end{figure*}
+
+\clearpage
+
+\subsection{Block Header Transition}
+\label{sec:block-header-trans}
+
+\begin{todo}
+  Introduce $\leteq$ notation for let bindings and
+  switch entire document to using it.
+\end{todo}
+
+\begin{figure}[ht]
+  \emph{Block header environments}
+  \begin{equation*}
+    \BHeaderEnv =
+    \left(
+      \begin{array}{r@{~\in~}lr}
+        \var{s_{now}} & \Slot & \text{current slot number} \\
+        \var{pps} & \PParams & \text{Protocol parameters} \\
+        \var{sd} & \HashKey\mapsto\unitInterval & \text{relative stake distribution}\\
```

Comment:
doesn't have to be now, but I think it will be beneficial to define a stake type which isn't really the real unit interval, but effectively all `4.5 10^15` possible rational numbers.

Reply:
I agree, that's a great idea